### PR TITLE
Restyle Settings!

### PR DESCRIPTION
**Pull request by montehurd**
**Settings interface using standard grouped table bits!**
- [x] Setup rough scaffold for using SSSectionedDataSource (ss data source, cell model obj, cell obj and xib etc)
- [x] Make cell obj capable of displaying according to mock (icon, title, switch, disclosure icon, disclosure "EN" lang code text, etc) and able to accommodate long localized strings
- [x] Use the actual icons from the mocks
- [x] Use icon colors from mocks
- [x] Use localized strings
- [x] Update localized strings to match mocks
- [x] Hook up taps to load view controllers or external URLs
- [x] Hook up switch taps to do their things
- [x] Configure special functionality for cells/sections such login, search lang, and the debug section - i.e. make sure the old behavior works or works according to new mock 
- [x] Add "Support Wikimedia" option and hook it up!
- [x] Delete any cruft which is now completely orphaned as a result of this refactor
- [x] Use cell spacing from mocks
- [x] Use header spacing from mocks
- [x] Check with iOS 8 and adjust if needed
- [x] Figure out tap delay issue
- [x] Add visual tests for settings cells

untitled mov

### CODE REVIEW

**Comment by montehurd** [comment]
@bgerstle still getting delay on login tap - it's weird!

**Comment by montehurd** [comment]
@coreyfloyd this is ready for some reviewz... still not sure why tapping login has a delay. Ideas?

**Comment by nirzar** [comment]
@montehurd can we change the string "Rate app" to "Rate the app" and "Support Wikimedia" to "Support Wikipedia"

**Comment by montehurd** [comment]
@nirzar yes

**Comment by montehurd** [comment]
@bgerstle @coreyfloyd Should be good to go! The weird tap delay is fixed!

**Comment by montehurd** [comment]
@bgerstle @coreyfloyd Good to go?

**Comment by montehurd** [comment]
untitled mov

**Comment by montehurd** [comment]
untitled mov

File: Wikipedia/Code/WMFSettingsViewController.m [hunk 1]
```diff
@@ -1,634 +1,296 @@
...
+        cell.title          = menuItem.title;
+        cell.iconColor      = menuItem.iconColor;
+        cell.iconName       = menuItem.iconName;
+        cell.disclosureType = menuItem.disclosureType;
+        cell.disclosureText = menuItem.disclosureText;
+        
+        cell.selectionStyle = (menuItem.disclosureType == WMFSettingsMenuItemDisclosureType_Switch) ? UITableViewCellSelectionStyleNone : UITableViewCellSelectionStyleDefault;
+        
+        @strongify(self)
+        [cell.disclosureSwitch setOn:[self getSwitchOnValueForMenuItemType:menuItem.type]];
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFSettingsViewController.m]: Its better if this piece of information should be inside the WMFSettingsItem that way each cell is driven solely by the values stored in the model object.

**Comment by coreyfloyd** [comment]
:+1:

**Comment by bgerstle** [comment]
:+1:

**Pull request merged**